### PR TITLE
Created auth flow

### PR DESCRIPTION
**Pull request by zpawn**
Story https://app.shortcut.com/deskpro/story/93944/gitlab-auth

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+export type TicketData = {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    env: any,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    app: any,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ticket: any,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    currentAgent: any,
```
[the remark this answers is not in the document]
Comment (zpawn) [reply src/types.ts:18]: I don't need them now. In the next PR I will make entities associations and add types

**Review by iTweeno — APPROVED** [review]

File: src/types.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (iTweeno) [reply src/types.ts:18]: Alright, pass then

[the remark this answers is not in the document]
Comment (zpawn) [reply src/types.ts:18]: thanks David

File: manifest.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
   },
   "proxy": {
     "whitelist": [{
-      "url": "https://example.com/.*",
+      "url": "https://gitlab.com/.*",
```
Comment (AshleyDawson) [inline manifest.json:46]: Settings can now be injected into this url pattern, see https://support.deskpro.com/en-US/guides/developers/app-proxy#undefined-1

E.g.

```
  "url": "https://__instance_hostname_/.*"
```

**Review by AshleyDawson — APPROVED** [review]

File: manifest.json (hunk 2 above)
Comment (zpawn) [reply manifest.json:46]: I'll configure it in this story https://app.shortcut.com/deskpro/story/93951/gitlab-write-setup-guide-and-description

**Pull request merged**